From ca657c404dc8d8e945a6d30cae374eb7819c544d Mon Sep 17 00:00:00 2001
From: Nemanja Boric <nemanja.boric@sociomantic.com>
Date: Thu, 26 Jan 2017 18:52:49 +0100
Subject: [PATCH 15/19] Add bindings for POSIX getdelim and getline

---
 src/core/sys/posix/stdio.d | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/core/sys/posix/stdio.d b/src/core/sys/posix/stdio.d
index 122cba0..6cc30a9 100644
--- a/src/core/sys/posix/stdio.d
+++ b/src/core/sys/posix/stdio.d
@@ -321,3 +321,7 @@ unittest
     assert(memcmp(ptr, testdata.ptr, testdata.length*wchar_t.sizeof) == 0);
     assert(fclose(f) == 0);
 }
+
+
+ssize_t getdelim (char** lineptr, size_t* n, int delimiter, FILE* stream);
+ssize_t getline (char** lineptr, size_t* n, FILE* stream);
-- 
2.14.1

